test/integration/wan_lan_dht_test.go

```diff
@@ -0,0 +1,277 @@
+package integrationtest
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"math"
+	"math/rand"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-ipfs/core"
+	mock "github.com/ipfs/go-ipfs/core/mock"
+	libp2p2 "github.com/ipfs/go-ipfs/core/node/libp2p"
+
+	corenet "github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peerstore"
+	testutil "github.com/libp2p/go-libp2p-testing/net"
+	mocknet "github.com/libp2p/go-libp2p/p2p/net/mock"
+
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+func TestDHTConnectivityFast(t *testing.T) {
+	conf := testutil.LatencyConfig{
+		NetworkLatency:    0,
+		RoutingLatency:    0,
+		BlockstoreLatency: 0,
+	}
+	if err := RunDHTConnectivity(conf, 5); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestDHTConnectivitySlowNetwork(t *testing.T) {
+	SkipUnlessEpic(t)
+	conf := testutil.LatencyConfig{NetworkLatency: 400 * time.Millisecond}
+	if err := RunDHTConnectivity(conf, 5); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestDHTConnectivitySlowRouting(t *testing.T) {
+	SkipUnlessEpic(t)
+	conf := testutil.LatencyConfig{RoutingLatency: 400 * time.Millisecond}
+	if err := RunDHTConnectivity(conf, 5); err != nil {
+		t.Fatal(err)
+	}
+}
+
+var wanPrefix = net.ParseIP("100::")
+var lanPrefix = net.ParseIP("fe80::")
+
+func makeAddr(n uint32, wan bool) ma.Multiaddr {
+	var ip net.IP
+	if wan {
+		ip = append(net.IP{}, wanPrefix...)
+	} else {
+		ip = append(net.IP{}, lanPrefix...)
+	}
+
+	binary.LittleEndian.PutUint32(ip[12:], n)
+	addr, _ := ma.NewMultiaddr(fmt.Sprintf("/ip6/%s/tcp/4242", ip))
+	return addr
+}
+
+func RunDHTConnectivity(conf testutil.LatencyConfig, numPeers int) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// create network
+	mn := mocknet.New(ctx)
+	mn.SetLinkDefaults(mocknet.LinkOptions{
+		Latency:   conf.NetworkLatency,
+		Bandwidth: math.MaxInt32,
+	})
+
+	testPeer, err := core.NewNode(ctx, &core.BuildCfg{
+		Online: true,
```

Comment:
We should start the test peer as a client. Otherwise, the test record could get put on the test peer.